Networking/Networking/Model/SystemPlugin.swift

```diff
@@ -0,0 +1,91 @@
+import Foundation
+import Codegen
+
+public struct SystemPlugin: Decodable, GeneratedFakeable, GeneratedCopiable {
+    public let siteID: Int64
+    public let plugin: String           // e.g. woocommerce/woocommerce.php
+    public let name: String             // e.g. WooCommerce
+    public let version: String          // e.g. 3.0.0-rc.1
+    public let versionLatest: String    // e.g. 2.6.14
+    public let url: String              // e.g. https://woocommerce.com/
+    public let authorName: String       // e.g. Automattic
+    public let authorUrl: String        // e.g. https://woocommerce.com
+    public let networkActivated: Bool   // i.g false | true
+
+    /// Struct initializer.
+    ///
+    public init(siteID: Int64,
+                plugin: String,
+                name: String,
+                version: String,
+                versionLatest: String,
+                url: String,
+                authorName: String,
+                authorUrl: String,
+                networkActivated: Bool) {
+        self.siteID = siteID
+        self.plugin = plugin
+        self.name = name
+        self.version = version
+        self.versionLatest = versionLatest
+        self.url = url
+        self.authorName = authorName
+        self.authorUrl = authorUrl
+        self.networkActivated = networkActivated
+    }
+
+    /// The public initializer for SystemPlugin.
+    ///
+    public init(from decoder: Decoder) throws {
+        guard let siteID = decoder.userInfo[.siteID] as? Int64 else {
+            throw SystemPluginError.missingSiteID
+        }
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let plugin = try container.decode(String.self, forKey: .plugin)
+        let name = try container.decodeIfPresent(String.self, forKey: .name) ?? ""
+        let version = try container.decodeIfPresent(String.self, forKey: .version) ?? ""
+        let versionLatest = try container.decodeIfPresent(String.self, forKey: .versionLatest) ?? ""
+        let url = try container.decodeIfPresent(String.self, forKey: .url) ?? ""
+        let authorName = try container.decodeIfPresent(String.self, forKey: .authorName) ?? ""
+        let authorUrl = try container.decodeIfPresent(String.self, forKey: .authorUrl) ?? ""
+        let networkActivated = try container.decodeIfPresent(Bool.self, forKey: .networkActivated) ?? false
```

Comment:
Why are you using `decodeIfPressent` here? Have you found scenarios where those fields are optional?

Reply:
Well, since I don't know all the scenarios, I decided to use `decodeIfPressent` to avoid any kind of crash here. In the end, I don't know if the service could return a semi-empty object.
If you know that we won't have this scenario, I will change it!

Reply:
Using `decode` won't crash because this function `throws`, so the error should be captured and logged in the consumer layer.

As those fields should always exist, I think it's best to guarantee the entity integrity and provide only the ones that we consider complete and valid.

Reply:
You are right. I've changed it and not looks better.